velox/functions/prestosql/aggregates/AverageAggregate.cpp

```diff
@@ -465,66 +63,66 @@ exec::AggregateRegistrationResult registerAverage(const std::string& name) {
           switch (inputType->kind()) {
             case TypeKind::SMALLINT:
               return std::make_unique<
-                  AverageAggregate<int16_t, double, double>>(resultType);
+                  AverageAggregateBase<int16_t, double, double>>(resultType);
             case TypeKind::INTEGER:
               return std::make_unique<
-                  AverageAggregate<int32_t, double, double>>(resultType);
+                  AverageAggregateBase<int32_t, double, double>>(resultType);
             case TypeKind::BIGINT: {
               if (inputType->isShortDecimal()) {
-                return std::make_unique<DecimalAverageAggregate<int64_t>>(
+                return std::make_unique<DecimalAverageAggregateBase<int64_t>>(
                     resultType);
               }
               return std::make_unique<
-                  AverageAggregate<int64_t, double, double>>(resultType);
+                  AverageAggregateBase<int64_t, double, double>>(resultType);
             }
             case TypeKind::HUGEINT: {
               if (inputType->isLongDecimal()) {
-                return std::make_unique<DecimalAverageAggregate<int128_t>>(
+                return std::make_unique<DecimalAverageAggregateBase<int128_t>>(
                     resultType);
               }
               VELOX_NYI();
             }
             case TypeKind::REAL:
-              return std::make_unique<AverageAggregate<float, double, float>>(
-                  resultType);
+              return std::make_unique<
+                  AverageAggregateBase<float, double, float>>(resultType);
             case TypeKind::DOUBLE:
-              return std::make_unique<AverageAggregate<double, double, double>>(
-                  resultType);
+              return std::make_unique<
+                  AverageAggregateBase<double, double, double>>(resultType);
             default:
               VELOX_FAIL(
                   "Unknown input type for {} aggregation {}",
                   name,
                   inputType->kindName());
           }
         } else {
-          checkSumCountRowType(
+          checkSumCountRowTypeForAvg(
               inputType,
               "Input type for final aggregation must be (sum:double/long decimal, count:bigint) struct");
```

Comment:
This error message appears to be the same in both Presto and Spark. If that's the case, then we can move it inside checkSumCountRowTypeForAvg function and remove it from here. What do you think?

Also, I wonder if it makes sense to move checkSumCountRowTypeForAvg function to AverageAggregateBase.h since it part of avg implementation.

Reply:
Yes, moved to AverageAggregateBase.h and AverageAggregateBase.cpp for its implementation.